src/OpenTelemetry.Exporter.OneCollector/Internal/EventNameManager.cs

```diff
@@ -7,34 +7,45 @@
 
 namespace OpenTelemetry.Exporter.OneCollector;
 
+public enum EventNameDelimiter
+{
+    Period,
+    Underscore
+}
+
 internal sealed class EventNameManager
 {
     // Note: OneCollector will silently drop events which have a name less than 4 characters.
     internal const int MinimumEventFullNameLength = 4;
     internal const int MaximumEventFullNameLength = 100;
+
+    // Make change to allow underscore in eventnamespace below.
     private static readonly Regex EventNamespaceValidationRegex = new(@"^[A-Za-z](?:\.?[A-Za-z0-9]+?)*$", RegexOptions.Compiled);
     private static readonly Regex EventNameValidationRegex = new(@"^[A-Za-z][A-Za-z0-9]*$", RegexOptions.Compiled);
 
     private readonly string defaultEventNamespace;
     private readonly string defaultEventName;
+    private readonly EventNameDelimiter defaultEventNameDelimiter;
```

Comment:
I think this should just be 'eventNameDelimiter'. 
The 'defaultEventNamespace' and 'defaultEventName' properties are about the _values for a default event_, not the _default values for an arbitrary event_

Reply:
That is - someone could call log(dataFields: foo) without passing an event name, and the event name would be the default event name. (pseudocode, something like that)